pkg/apis/batch/types.go

```diff
@@ -110,6 +110,99 @@ const (
 	IndexedCompletion CompletionMode = "Indexed"
 )
 
+// PodFailurePolicyAction specifies how a Pod failure is handled.
+type PodFailurePolicyAction string
+
+const (
+
+	// This is an action which might be taken on pod failure - mark the
+	// pod's job as Failed and terminate all running pods.
+	PodFailurePolicyActionTerminate PodFailurePolicyAction = "Terminate"
+
+	// This is an action which might be taken on pod failure - the .backoffLimit is not incremented
+	// and a replacement pod is created.
+	PodFailurePolicyActionIgnore PodFailurePolicyAction = "Ignore"
```

Comment:
Hm, if we introduce it I would call it `Default`, but `Count` is also fine :). However, as we only consider `ReachedResourceLimit` (or a similar one added by Kubelet) for Beta, maybe then we could then introduce `Count` or `Default`. It does not seem to be needed if we have only a single condition as currently,

Reply:
The problem of introducing `Count` at the same time as `ReachedResourceLimit`, is that you have to wait one release until being able to enable the feature. This is because there could be an apiserver running an old version that would reject `Count` because it doesn't know about it.

Reply:
I see, it sort of makes sense already, given users are able to use custom condition codes.

Implemented and covered with unit tests. Haven't added an integration tests as these execute longer (~7s each) and the unit tests give 100% coverage of the code to support it.